feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go

```diff
@@ -257,30 +320,53 @@ func testBackupDecap(ctx context.Context, t *testing.T, args *testArgs) {
+	args.client.AddNHG(t, NHG100ID, map[uint64]uint64{NH100ID: 1}, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB)
+
+	t.Logf("Adding NHG %d NH %d with  %v  and backup NHG %d via gRIBI", NHG101ID, NH101ID, VIP1, NHG100ID)
+	args.client.AddNH(t, NH101ID, VIP1, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB)
+	args.client.AddNHG(t, NHG101ID, map[uint64]uint64{NH101ID: 1}, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB, &gribi.NHGOptions{BackupNHG: NHG100ID})
+
+	t.Logf("Adding an IPv4Entry for %s in VRF %s with primary VIP1, backup as VRF %s  via gRIBI", outerDstIP1, vrf1, vrf2)
+	args.client.AddIPv4(t, outerDstIP1+"/"+mask, NHG101ID, vrf1, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB)
 
-	t.Logf("Adding NH %d as decap and NHG %d via gRIBI", NH2ID, NH2ID)
-	args.client.AddNH(t, NH2ID, "Decap", *deviations.DefaultNetworkInstance, fluent.InstalledInFIB)
-	args.client.AddNHG(t, NH2ID, map[uint64]uint64{NH2ID: 100}, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB)
+	t.Logf("Adding NHG %d with NH %d as decap and DEFAULT vrf lookup via gRIBI", NHG103ID, NH103ID)
+	args.client.AddNH(t, NH103ID, "Decap", *deviations.DefaultNetworkInstance, fluent.InstalledInFIB, &gribi.NHOptions{VrfName: *deviations.DefaultNetworkInstance})
+	args.client.AddNHG(t, NHG103ID, map[uint64]uint64{NH103ID: 1}, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB)
 
-	t.Logf("Adding NH %d as DecapEncap and NHG %d via gRIBI", NH1ID, NH1ID)
-	args.client.AddNH(t, NH1ID, "DecapEncap", *deviations.DefaultNetworkInstance, fluent.InstalledInRIB, &gribi.NHOptions{Src: secondaryTunnelSrcIP, Dest: secondaryTunnelDstIP, VrfName: vrfName})
-	args.client.AddNHG(t, NH1ID, map[uint64]uint64{NH1ID: 100}, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB, &gribi.NHGOptions{BackupNHG: NH2ID})
-	t.Logf("Adding an IPv4Entry for %s with primary DecapEncap & backup decap via gRIBI", primaryTunnelDstIP)
-	args.client.AddIPv4(t, primaryTunnelDstIP+"/"+mask, NH1ID, vrfName, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB)
+	t.Logf("Adding NHG %d NH %d and  NH as decap and encap with destination vrf as %v and backup NHG %d via gRIBI", NHG102ID, NH102ID, vrf3, NHG103ID)
+	args.client.AddNH(t, NH102ID, "DecapEncap", *deviations.DefaultNetworkInstance, fluent.InstalledInFIB, &gribi.NHOptions{Src: outerSrcIP2, Dest: outerDstIP2, VrfName: vrf3})
+	args.client.AddNHG(t, NHG102ID, map[uint64]uint64{NH102ID: 1}, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB, &gribi.NHGOptions{BackupNHG: NHG103ID})
+
+	t.Logf("Adding an IPv4Entry for %s in vrf %s with decap and encap destiantion  in  %s via gRIBI", outerDstIP1, vrf2, vrf3)
+	args.client.AddIPv4(t, outerDstIP1+"/"+mask, NHG102ID, vrf2, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB)
+
+	t.Logf("Adding NHG %d NH %d and  NH as %v  and backup NHG %d via gRIBI", NHG104ID, NH104ID, VIP2, NHG103ID)
```

Comment:
Could you please move the programming of NH104, NHG104 and outerDstIP2 (L#364 - L#369) before programming NH102 (L#356) ? This is to ensure "Dest: outerDstIP2" is programmed before using it.

Reply:
Corrected